src/extension/importer/core/src/main/java/org/geoserver/importer/Importer.java

```diff
@@ -1768,14 +1953,16 @@ Throwable copyFromFeatureReader(
         int cnt = 0;
         // metrics
         long startTime = System.currentTimeMillis();
-        task.clearMessages();
 
-        task.setTotalToProcess(format.getFeatureCount(task.getData(), task));
+        try {
+            task.clearMessages();
+            task.setTotalToProcess(format.getFeatureCount(task.getData(), task));
+        } catch (IOException noCountAvailable) {
```

Comment:
The code is calling a ``FeatureSource.getCount(...)`` which might fail for a lot of reasons, not just for a "no count available". In other cases the exception is going to be fatal and indicating a data access issue (files gone, user no longer able to access resources, and so on). Seems like something that sould be handled in the implemnetation of the feature source instead (like falling back on interating, like ContentFeatureSource does).

Reply:
I did not look deeply into this part and was doing my best not to change the design (viewing the code as stable if not pretty).

Still if I understand your feed I may of introduced a regression here and should re-throw the exception to preserver prior execution path.

Reply:
Reviewed the logic and it had gaps before I started; adjust to keep the error and allow the " cancel or failure, rollback transaction" code at the end to cleanup